go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -235,6 +235,15 @@ func (bsnc *BaseSecondaryNetworkController) ensurePodForSecondaryNetwork(pod *ka
 		return err
 	}
 
+	if bsnc.IsPrimaryNetwork() && (bsnc.TopologyType() == "layer2" || bsnc.TopologyType() == "layer3") {
+		network := &nadapi.NetworkSelectionElement{
+			// TODO: we need to get the NAD name here, not the network name.
+			Name:      bsnc.GetNetworkName(),
+			Namespace: pod.Namespace,
+		}
+		return bsnc.ensurePodForUserDefinedPrimaryNet(pod, network)
+	}
+
 	on, networkMap, err := util.GetPodNADToNetworkMapping(pod, bsnc.NetInfo)
```

Comment:
an alternative would be to set the `dummy` NSE created above in this map. Then it would just munch it and spit out the LS+LSP for the user-defined primary network.